src/sentry/grouping/ingest/seer.py

```diff
@@ -241,16 +243,49 @@ def maybe_check_seer_for_matching_grouphash(event: Event) -> GroupHash | None:
             sample_rate=options.get("seer.similarity.metrics_sample_rate"),
             tags={"call_made": True, "blocker": "none"},
         )
+
         try:
             # If no matching group is found in Seer, we'll still get back result
             # metadata, but `seer_matched_grouphash` will be None
             seer_response_data, seer_matched_grouphash = get_seer_similar_issues(event)
-            event.data["seer_similarity"] = seer_response_data
-
-        # Insurance - in theory we shouldn't ever land here
-        except Exception as e:
+        except Exception as e:  # Insurance - in theory we shouldn't ever land here
             sentry_sdk.capture_exception(
                 e, tags={"event": event.event_id, "project": event.project.id}
             )
+            return None
+
+        # Find the GroupHash corresponding to the hash value sent to Seer
+        #
+        # TODO There shouldn't actually be more than one hash here, but
+        #   a) there's a bug in our precedence logic which leads both in-app and system stacktrace
+        #      hashes being marked as contributing and making it through to this point, and
+        #   b) because of how we used to compute secondary and primary hashes, we keep secondary
+        #      hashes even when we don't need them.
+        # Once those two problems are fixed, there will only be one hash passed to this function
+        # and we won't have to do this search to find the right one to update.
+        primary_hash = event.get_primary_hash()
+        grouphash_sent = list(
+            filter(lambda grouphash: grouphash.hash == primary_hash, all_grouphashes)
+        )[0]
+
+        # Update the relevant GroupHash with Seer results
+        metadata = grouphash_sent.metadata
+        if metadata:
+            # Technically the time of the metadata record creation and the time of the Seer
+            # request will be some milliseconds apart, but the difference isn't meaningful
+            # and forcing them to be the same (rather than just close) lets us use their
+            # equality as a signal that the Seer call happened during ingest rather than
+            # during a backfill, without having to store that information separately
+            metadata.seer_date_sent = metadata.date_added
+            metadata.seer_event_sent = event.event_id
+            metadata.seer_model = seer_response_data["similarity_model_version"]
+            metadata.seer_matched_grouphash = seer_matched_grouphash
+            metadata.seer_match_distance = (
+                seer_response_data["results"][0]["stacktrace_distance"]
+                if seer_matched_grouphash
+                else None
+            )
+
+            metadata.save()
```

Comment:
Nit: It can be helpful to include `update_fields` in your save so that we only modify the affected columns, or alternatively use `metadata.update(seer_date_sent=metadata.date_added, ...)`

Reply:
Good idea. I'd originally shied away from `.update()` because it seemed like there were going to be multiple records updated and I was going to try to do it in bulk. Now that it's only one record, though, you're right that there's no loss in using it. I'll switch it.